sdk/messaging/azservicebus/internal/token_provider.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package internal
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/Azure/azure-amqp-common-go/v3/auth"
+	"github.com/Azure/azure-amqp-common-go/v3/sas"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+)
+
+// TokenProvider handles access tokens and expiration calculation for SAS
+// keys (via connection strings) or TokenCredentials from Azure Identity.
+type TokenProvider struct {
```

Comment:
Does this type need to be exported?  Looks like you can only create it internally.

Reply:
Done.